usaspending_api/agency/v2/views/obligations_by_award_category.py

```diff
@@ -0,0 +1,76 @@
+from rest_framework.request import Request
+from rest_framework.response import Response
+from typing import Any
+
+from usaspending_api.agency.v2.views.agency_base import AgencyBase
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.elasticsearch.filter_helpers import create_fiscal_year_filter
+from usaspending_api.common.elasticsearch.search_wrappers import TransactionSearch
+from usaspending_api.common.query_with_filters import QueryWithFilters
+
+
+class ObligationsByAwardCategory(AgencyBase):
+    """
+    Returns a breakdown of obligations by award category (contracts, IDVs, grants, loans, direct payments, other)
+    within the requested fiscal year (or current FY).
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/agency/toptier_code/obligations_by_award_category.md"
+
+    @cache_response()
+    def get(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+
+        results = self.query_elasticsearch()
+
+        formatted_results = self.format_results(results)
+
+        return Response(formatted_results)
+
+    def query_elasticsearch(self):
+        filters = {}
+        filters["time_period"] = create_fiscal_year_filter(self.fiscal_year)
+        filters["agencies"] = [{"toptier_id": self.agency_id, "type": "awarding", "tier": "toptier"}]
+
+        filter_query = QueryWithFilters.generate_transactions_elasticsearch_query(filters)
+        search = TransactionSearch().filter(filter_query)
+
+        search.aggs.bucket("group_by_award_category", "terms", field="award_category").metric(
+            "obligations", "sum", field="generated_pragmatic_obligation"
```

Comment:
Just to confirm - the `generated_pragmatic_obligation` is a coalesce of both `loan_subsidy_cost` and `federal_action_obligation`, correct? When testing locally I'm not seeing any values for loans, which might be because of my local db, but I just wanted to double check

Reply:
Yep, from `transaction_search.sql`: 

>   COALESCE(CASE
WHEN transaction_normalized.type IN('07','08') THEN transaction_normalized.original_loan_subsidy_cost
    ELSE transaction_normalized.federal_action_obligation
  END, 0)::NUMERIC(23, 2) AS generated_pragmatic_obligation